lightwood/data/timeseries_analyzer.py

```diff
@@ -56,5 +68,35 @@ def get_delta(df: pd.DataFrame, ts_info: dict, group_combinations: list, order_c
                             lambda x: x.iloc[1] - x.iloc[0])
                         delta = rolling_diff.value_counts(ascending=False).keys()[0]
                         deltas[group][col] = delta
+        ts_info['data'] = original_data
 
     return deltas
+
+
+def get_naive_residuals(target_data: pd.DataFrame, m: int = 1) -> Tuple[List, float]:
+    """
+    Computes forecasting residuals for the naive method (forecasts for time `t` is the value observed at `t-1`).
+    Useful for computing MASE forecasting error.
+
+    Note: method assumes predictions are all for the same group combination. For a dataframe that contains multiple
+     series, use `get_grouped_naive_resiudals`.
+
+    m: season length. the naive forecasts will be the m-th previously seen value for each series
```

Comment:
Change to:

:param target_data: The input sequence for forecasting <might be wrong?>
:param m: season length; naive forecasts will be the m-th previously seen value for each series

:returns: <whatever it returns>